sematic/ui/packages/common/src/pages/RunDetails/contextMenus/FunctionSectionMenu.tsx

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useContext, useMemo } from "react";
+import ContextMenu from "src/component/ContextMenu";
+import { useRootRunContext } from "src/context/RootRunContext";
+import { useRunDetailsSelectionContext } from "src/context/RunDetailsSelectionContext";
+import SnackBarContext from "src/context/SnackBarContext";
+import { useCancelRun, useRerun } from "src/hooks/resolutionHooks";
+
+interface FunctionSectionActionMenuProps {
+    anchorEl: HTMLElement | null;
+}
+
+function FunctionSectionActionMenu(props: FunctionSectionActionMenuProps) {
+    const { anchorEl } = props;
+
+    const { rootRun, resolution, graph } = useRootRunContext();
+    const { selectedRun  } = useRunDetailsSelectionContext();
+
+    const { setSnackMessage } = useContext(SnackBarContext);
+
+    const [rerunState, rerun] = useRerun(rootRun?.id, selectedRun?.id);
+    const [cancelRunState, cancel] = useCancelRun(rootRun!.id);
+
+    const selectedRunInputEdges = useMemo(
+        () => graph?.edges.filter((edge) => edge.destination_run_id === selectedRun?.id),
+        [graph, selectedRun?.id]
+    );
+
+    const cancelEnabled = useMemo(
+        () => {
+            if (!rootRun) {
+                return false;
```

Comment:
Why is this?
We could have a long-running one-func standalone pipeline that we would want to be able to cancel.

Reply:
This is to null-guard the read operation on line 37.

Is it a mistake? Maybe we should look at `selectedRun.future_state`? Can you help confirm?

(The old UI doesn't have this command, I don't know what to reference)

@tscurtu

Reply:
Looking at this again, I believe I must have misunderstood something.
Please ignore the comment!